core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -81,6 +81,7 @@ public Transaction newReplaceTableTransaction(
       String location,
       Map<String, String> properties,
       boolean orCreate) {
+    Preconditions.checkArgument(isValidIdentifier(identifier), "Invalid table identifier: %s", identifier);
```

Comment:
I think that this is already done because it uses the table builder and that has the same check on line 201.

Reply:
Yes. Missed that. this line is not needed.